server/src/room_matchmaker.cpp

```diff
@@ -2,37 +2,53 @@
 #include <algorithm>
 #include <cassert>
 #include "../include/game_handler.h"
+#include "../include/server.h"
 
 namespace war_of_ages::server {
 room_matchmaker &room_matchmaker::instance() {
     static room_matchmaker inst;
     return inst;
 }
 
-bool room_matchmaker::user_exists(const std::string &handle) const {
-    std::unique_lock l(m_mutex_status);
-    return m_users_in_rooms.find(handle) != m_users_in_rooms.end();
-}
-
 bool room_matchmaker::add_user_to_room(const std::string &handle, const std::string &room_id) {
     std::unique_lock l(m_mutex_status);
-    if (user_exists(handle)) {
+    if (m_users_in_rooms.count(handle)) {
         return false;
     }
+    message<messages_type> msg;
+    msg.header.id = messages_type::ROOM_JOIN_RESPONSE;
     auto &room = m_rooms[room_id];
     if (room.size() == 2) {
+        msg << static_cast<std::uint8_t>(0);
+        server::instance().send_message(handle, msg);
         return false;
     }
     room.push_back(handle);
     m_users_in_rooms.insert({handle, {room_id, user_in_room::user_status::NOT_READY}});
+
+    if (room.size() == 2) {
+        msg << static_cast<std::uint8_t>(0)
+            << static_cast<std::uint8_t>(
+                   m_users_in_rooms[room[0]].status == user_in_room::user_status::READY ? 1 : 0)
+            << room[0];
+
+        message<messages_type> msg2;
+        msg2.header.id = messages_type::ROOM_ENEMY_JOINED;
+        msg2 << static_cast<std::uint8_t>(0) << static_cast<std::uint8_t>(0) << handle;
+        server::instance().send_message(room[0], msg2);
+    } else {
+        msg << static_cast<std::string>("");
```

Comment:
```suggestion
        msg << std::string{};
```
а если сделано using namespace std::literals, то и вовсе << ""s